### PR TITLE
Use a valid list retriever for the global Update() function

### PR DESCRIPTION
**Pull request by toddgaunt-gs**
The last update to the list retriever, in order to use a custom client, passed the tests provided but was still broken since it was using a nil http.Client for the provided Update() function which broke the library. This change fixes this issue, and adds an integration test to at least call Update() to ensure that it doesn't get silently broken again.

### CODE REVIEW

**Review by mara-soldan — APPROVED** [review]

File: listretriever.go [hunk 1]
```diff
@@ -81,7 +81,13 @@ func (gh gitHubListRetriever) GetLatestReleaseTag() (string, error) {
 func (gh gitHubListRetriever) GetList(release string) (io.Reader, error) {
 	var url = fmt.Sprintf(publicSuffixURL, release)
 
-	var res, err = gh.client.Get(url)
+	// Just in case a nil client was passed, use the default http client.
+	client := http.DefaultClient
```
Comment (angela-globalsign) [inline listretriever.go:85]: This might be needed in the GetLatestReleaseTag()  function as well.

Comment (angela-globalsign) [reply listretriever.go:85]: Ignore me your passing it in UpdateWithListRetriever

Comment (angela-globalsign) [reply listretriever.go]: I am confused why do you need this for GetList() and not for GetLatestReleaseTag()

Comment (toddgaunt-gs) [reply listretriever.go]: Good catch! I just updated it. Also added a method to retrieve the client.

Comment (toddgaunt-gs) [reply listretriever.go]: Just a safeguard!

**Pull request merged**